packages/bff/src/routers/agreementRouter.ts

```diff
@@ -28,9 +84,26 @@ const agreementRouter = (
     .get("/agreements/filter/consumers", async (_req, res) =>
       res.status(501).send()
     )
-    .get("/agreements/:agreementId", async (_req, res) =>
-      res.status(501).send()
-    )
+
+    .get("/agreements/:agreementId", async (req, res) => {
+      const ctx = fromBffAppContext(req.ctx, req.headers);
+
+      try {
+        const result = await agreementService.getAgreementById(
+          req.params.agreementId,
+          ctx
+        );
+        return res.status(200).json(result).end();
+      } catch (error) {
+        const errorRes = makeApiProblem(
+          error,
+          getAgreementsErrorMapper,
```

Comment:
Yes that may be better, only thing is: i cannot define two identical functions because `eslint/sonarjs` complains. In this case, should i do it and simply disable that rule inline?

Reply:
Yeah for cases like this let's disable the rule inline :)